App/i18n/index.ts

```diff
@@ -0,0 +1,27 @@
+import i18n from 'i18next'
+import { initReactI18next } from 'react-i18next'
+import * as RNLocalize from 'react-native-localize'
+import { defaultLanguage } from '../constants'
+import en from './en'
+
+const resources = {
+  en: {
+    translation: en,
+  },
+}
```

Comment:
Add a type `Translation` that is based on the `en` translation. Other languages can now set this as the type, meaning that if `en` adds a property we will get an error if it is not added for other languages.


```suggestion
export type Translation = typeof en
```

Then create a `types/react-i18next.d.ts` file and add the following to it:

```
import { Translation } from '../src/translations' # update with your path

import 'react-i18next'

declare module 'react-i18next' {
  interface CustomTypeOptions {
    resources: {
      translation: Translation
    }
  }
}
```

The `t()` method from `useTranslation()` is now fully typed